var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -0,0 +1,347 @@
+            description='Compile with SuiteSparse solvers')
+    # not everyone has py-numpy activated, keep it disabled by default to avoid
+    # configure errors
+    variant('python',       default=False, description='Build python wrappers')
+    variant('shared',       default=True,
+            description='Enables the build of shared libraries')
+    variant('debug',        default=False,
+            description='Builds a debug version of the libraries')
+    variant('boost',        default=True, description='Compile with Boost')
+
+    depends_on('cmake', type='build')
+
+    # Everything should be compiled with -fpic
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('boost', when='+boost')
+    depends_on('matio')
+    depends_on('glm')
+    depends_on('metis@5:', when='+metis')
+    depends_on('suite-sparse', when='+suite-sparse')
+
+    # MPI related dependencies
+    depends_on('mpi')
+    depends_on('netcdf+mpi')
+    depends_on('parmetis', when='+metis')
+    # Trilinos' Tribits config system is limited which makes it very tricky to
+    # link Amesos with static MUMPS, see
+    # https://trilinos.org/docs/dev/packages/amesos2/doc/html/classAmesos2_1_1MUMPS.html
+    # One could work it out by getting linking flags from mpif90 --showme:link
+    # (or alike) and adding results to -DTrilinos_EXTRA_LINK_FLAGS together
+    # with Blas and Lapack and ScaLAPACK and Blacs and -lgfortran and it may
+    # work at the end. But let's avoid all this by simply using shared libs
+    depends_on('mumps@5.0:+mpi+shared', when='+mumps')
+    depends_on('scalapack', when='+mumps')
+    depends_on('superlu-dist@:4.3', when='@:12.6.1+superlu-dist')
+    depends_on('superlu-dist', when='@12.6.2:+superlu-dist')
+    depends_on('hypre~internal-superlu', when='+hypre')
+    depends_on('hdf5+mpi', when='+hdf5')
+    depends_on('python', when='+python')
+    depends_on('py-numpy', when='+python')
+    depends_on('swig', when='+python')
+
+    patch('umfpack_from_suitesparse.patch')
+
+    # check that the combination of variants makes sense
+    def variants_check(self):
+        if '+superlu-dist' in self.spec and self.spec.satisfies('@:11.4.3'):
+            # For Trilinos v11 we need to force SuperLUDist=OFF, since only the
+            # deprecated SuperLUDist v3.3 together with an Amesos patch is
+            # working.
+            raise RuntimeError('The superlu-dist variant can only be used' +
+                               ' with Trilinos @12.0.1:')
+
+    def install(self, spec, prefix):
+        self.variants_check()
+
+        cxx_flags = []
+        options = []
+        options.extend(std_cmake_args)
+
+        mpi_bin = spec['mpi'].prefix.bin
+        # Note: -DXYZ_LIBRARY_NAMES= needs semicolon separated list of names
+        blas = spec['blas'].blas_libs
+        lapack = spec['lapack'].lapack_libs
+        options.extend([
+            '-DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON',
```

Comment:
I would say yes, unless you want to add variants to `trilinos` for each available package. Or a few most commont plus `the-rest`. 

Otherwise dependent packages have no way to specify which parts of `trilinos` they need.